### DISCUSSION TITLE
Understanding model accuracy: MobileDet vs public YOLOv9 vs Frigate+

### DISCUSSION THREAD

**Discussion by ThorstenUm** [blakeblackshear/frigate#23982] [Ask A Question] [answered]
Hi everyone,

I would like to make sure I understand the Frigate model situation correctly.

I have read the documentation in several places, but I am still not sure I fully understand how the different model types relate to each other in terms of training data and detection accuracy.

Until now I have been using the default Coral / EdgeTPU setup, which I understand uses the included MobileDet model by default. On my new system I am testing an NVIDIA GPU with an ONNX YOLOv9 model.

My main goal is not to detect more exotic object classes. I mainly care about improving the reliability and accuracy for the important common objects, especially:

- person
- car
- bicycle
- motorcycle
- dog / cat

What I am trying to understand is:

1. Is the included Coral / EdgeTPU MobileDet default model newer or more actively updated than the public YOLOv9 models mentioned in the documentation, or is it simply the default because it is small and optimized for Coral?

2. When I generate or download a YOLOv9 ONNX model for Frigate, am I correct that this is based on fixed public pretrained weights and does not automatically benefit from newer Frigate training data?

3. Are Frigate+ models the only Frigate-provided models that receive newer training data over time, based on real security camera images from Frigate+ submissions?

4. If my goal is better detection quality for common objects like people and cars, rather than more object categories, is Frigate+ generally the recommended path compared with the default MobileDet model or a public YOLOv9 model?

5. Is there currently a recommended model type for best accuracy on NVIDIA GPUs, especially when using Frigate+ models? For example, should I prefer YOLOv9, RF-DETR, YOLO-NAS, or something else?

I am trying to decide whether moving from Coral/MobileDet to NVIDIA/YOLOv9 already gives me a meaningful accuracy improvement, or whether the main improvement would come from using Frigate+ base/fine-tuned models.

Thanks for clarifying!

**Comment by inkeep-ai-assistant** [comment]
Here's what the sources support:

**1. MobileDet default vs public YOLOv9**

The included TensorFlow Lite model is provided in the container at `/edgetpu_model.tflite` and is used by the Edge TPU detector type by default [(1)](https://docs.frigate.video/configuration/object_detectors). For EdgeTPU, Mobiledet is listed as the default model, and YOLOv9 is described as "More accurate but slower than default model" [(1)](https://docs.frigate.video/configuration/object_detectors). There's no documentation about relative update cadence of the bundled model versus public YOLOv9 weights, so I can't confirm one is "newer" than the other.

**2. Public YOLOv9 ONNX models**

Public YOLOv9 models are supported but not included by default [(1)](https://docs.frigate.video/configuration/object_detectors). There is nothing indicating these receive newer Frigate training data — the fine-tuning workflow described in the sources applies to Frigate+ models [(2)](https://docs.frigate.video/plus).

**3. Frigate+ and newer training data**

Yes. Frigate+ offers models trained on images submitted by Frigate+ users from their security cameras, specifically designed for the way Frigate NVR analyzes video footage. Images you upload fine tune a base model trained from images uploaded by all Frigate+ users [(2)](https://docs.frigate.video/plus). New base models are published quarterly with target dates of January 15th, April 15th, July 15th, and October 15th, and a subscription includes 12 model trainings per year [(2)](https://docs.frigate.video/plus).

**4. Better accuracy for common objects**

Frigate+ models "offer higher accuracy with less resources" [(2)](https://docs.frigate.video/plus). Note that some users may find Frigate+ models result in more false positives initially, but the model improves by submitting true and false positives — aim for roughly 80% true positive and 20% false positive submissions, and at least 100 verified images per camera across varied conditions [(2)](https://docs.frigate.video/plus). Only images with at least one verified label are used in training, and verification must be done in Frigate+ [(2)](https://docs.frigate.video/plus).

**5. NVIDIA GPU recommendation with Frigate+**

For NVidia GPU, the recommended detector type is `onnx` and the recommended model type is `yolov9` [(2)](https://docs.frigate.video/plus). YOLOv9 is described as a leading SOTA object detection model with similar performance to yolonas but on a wider range of hardware [(2)](https://docs.frigate.video/plus). On the ONNX detector, YOLOv9 supports CUDA Graphs for optimal Nvidia performance, while YOLO-NAS is not supported by CUDA Graphs [(1)](https://docs.frigate.video/configuration/object_detectors). RF-DETR also supports CUDA Graphs on Nvidia [(1)](https://docs.frigate.video/configuration/object_detectors).

Frigate+ offers three model types — `mobiledet`, `yolonas`, and `yolov9` — all trained to detect the same label set [(2)](https://docs.frigate.video/plus). RF-DETR is not listed as a Frigate+ model type [(2)](https://docs.frigate.video/plus).

If you use a Frigate+ model, do not define any of the `model` parameters in your config except for `path` [(1)](https://docs.frigate.video/configuration/object_detectors). A basic ONNX detector config looks like:

```
detectors:
  onnx:
    type: onnx
```
[(1)](https://docs.frigate.video/configuration/object_detectors)

Your target objects — `person`, `car`, `bicycle`, `motorcycle`, `dog`, `cat` — are all supported Frigate+ labels [(2)](https://docs.frigate.video/plus).

📚 **Sources:**
1. [Supported Hardware](https://docs.frigate.video/configuration/object_detectors)
2. [Models](https://docs.frigate.video/plus)

---

Have another question? Just tag [@inkeep](https://inkeep.com).

**Reply by hawkeye217** [reply]
Generally the AI is correct here, just a few confirming comments:

> Is the included Coral / EdgeTPU MobileDet default model newer or more actively updated than the public YOLOv9 models mentioned in the documentation, or is it simply the default because it is small and optimized for Coral?

YOLOv9 is going to perform better because it's a better model architecture than mobiledet. The public, non-Frigate+ model is not maintained by us.

> 2. When I generate or download a YOLOv9 ONNX model for Frigate, am I correct that this is based on fixed public pretrained weights and does not automatically benefit from newer Frigate training data?

Correct, no Frigate+ training data is included with the public YOLOv9 models.

> 3. Are Frigate+ models the only Frigate-provided models that receive newer training data over time, based on real security camera images from Frigate+ submissions?

Most of the public models are based on the COCO dataset, which does not change. Frigate+ is constantly evolving based on user submissions.

> If my goal is better detection quality for common objects like people and cars, rather than more object categories, is Frigate+ generally the recommended path compared with the default MobileDet model or a public YOLOv9 model?

Yes.

> 3. Is there currently a recommended model type for best accuracy on NVIDIA GPUs, especially when using Frigate+ models? For example, should I prefer YOLOv9, RF-DETR, YOLO-NAS, or something else?

YOLOv9 is generally the recommendation. RF-DETR has also proven to work well for some users, but inference times are generally higher.

**Reply by ThorstenUm** [reply]
Thanks, that helps a lot and confirms my understanding.

I have one follow-up question from a user perspective, and I hope it does not come across as criticism. I fully understand that Frigate+ needs to fund the training infrastructure and ongoing model improvements.

What I am still wondering is this:

Since Frigate+ already trains and provides Frigate-specific models for different hardware types, would it make sense to provide an older or less frequently updated Frigate-trained base model as the default free model for Frigate?

For example:

- Free Frigate:
  - one official Frigate-trained base model
  - maybe updated rarely or only with major releases
  - intended as the recommended default for common objects like person, car, bicycle, dog, cat

- Frigate+:
  - latest and continuously improved base models
  - access to new quarterly model updates
  - fine-tuning with the user’s own verified images
  - better accuracy and faster improvements over time

From my perspective, this would make the model situation much easier to understand. Right now, the free setup feels a bit confusing because users have to choose between MobileDet, public YOLOv9, YOLO-NAS, RF-DETR, different detector types, different hardware backends, and models from different sources.

I am not asking for the newest Frigate+ models to be made free. I am just trying to understand why Frigate does not use an older Frigate-trained base model as the official default, instead of relying on a collection of public/external models that are not trained on Frigate/security-camera data.

Is this mainly a licensing/data usage issue, a support burden, a technical maintenance issue across different hardware targets, or simply a product decision to keep Frigate+ as the place for Frigate-trained models?

Again, I do not mean this negatively. I am just trying to understand the reasoning, because for users the most important goal is usually not more object classes, but more reliable detection of the common important objects.

**Reply by hawkeye217** [reply]
I don't work on Frigate+ as it is exclusively Blake's business so this is probably a question best suited for him @blakeblackshear

**Reply by ThorstenUm** [reply]
Thanks for clarifying, that makes sense.

@blakeblackshear if you have a moment, I would be interested in your perspective on this.

I fully understand that Frigate+ needs to fund model training, infrastructure and ongoing improvements. My question is not meant as criticism of the subscription model.

I am mainly trying to understand the product reasoning:

Since Frigate+ already trains Frigate-specific models for different hardware types, would it make sense to provide an older or less frequently updated Frigate-trained base model as the official free/default model for Frigate, while keeping the latest base models, quarterly updates and custom fine-tuning as Frigate+ features?

From a user perspective, this might make the model situation easier to understand than choosing between several public/external models that are not trained on Frigate/security-camera data.

Is the reason mainly licensing/data usage, support burden, technical maintenance across hardware targets, or simply that Frigate-trained models are intentionally part of Frigate+?

Thanks!

**Reply by hawkeye217** [reply]
> From my perspective, this would make the model situation much easier to understand. Right now, the free setup feels a bit confusing because users have to choose between MobileDet, public YOLOv9, YOLO-NAS, RF-DETR, different detector types, different hardware backends, and models from different sources.

For a little clarity, MobileDet, YOLOv9, YOLO-NAS, and RF-DETR are model *architectures*, which are the shape of the network, not the thing that determines what it can recognize. Frigate+ doesn't add another architecture to that list, it changes the training data. The public models are trained on generic datasets like COCO, which are mostly internet photos taken at eye level, and they carry 80 labels you'll never care about. Frigate+ models are trained on images submitted from real security cameras, including your own, against Frigate's specific label set, so they're much better at the actual job of recognizing a person or car at 3am from a rooftop angle.

The reason you still pick an architecture with Frigate+ is that your hardware decides it for you, not your preference. A Coral can only run models compiled for the EdgeTPU. OpenVINO, TensorRT, ONNX, and Hailo each support a different set. So the flow is: figure out which model type your detector supports, then request that model type from Frigate+. Once you're on a Frigate+ model, you don't set width, height, input tensor, or labelmap in your config at all, just the path. That part actually gets simpler, not harder.

**Reply by blakeblackshear** [reply, marked as answer]
I don't have any plans to publish models trained on Frigate+ data for free. My view is that Frigate and Frigate+ are already well balanced between free and non-free functionality. Many users are satisfied with the COCO trained models for their needs, and we don't gatekeep core features behind a pay wall. There will likely be more efforts to simplify the set up and selection process for models in Frigate independent from Frigate+.

**Reply by ThorstenUm** [reply]
Thanks a lot, that clarification was very helpful.

I now understand the distinction much better: the model architecture is one part, but the training data is what really makes the difference for my use case. The explanation about COCO-trained public models versus Frigate+ models trained on real security camera images helped me put this into perspective.

I also appreciate the clear answer regarding the free/non-free balance. That makes sense.

I want to say again that I really appreciate the work and engagement that goes into Frigate and Frigate+. It is easy as a user to ask for more features or simpler defaults, but I also understand that training, maintaining and supporting good models takes real effort and resources.

Based on this discussion, I will use Frigate+ for my setup. The answers here helped me make that decision and understand the reasoning behind it.

Thanks again!
